packages/ui/src/accounts/hooks/useGroupLocks/useGroupLocks.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  isCandidateLockRecoverable,
+  isVoteLockRecoverable,
+  isWGLockRecoverable,
+  useLockOrientedAddressInformation,
+} from '@/accounts/hooks/useGroupLocks/helpers'
+import { BalanceLock } from '@/accounts/types'
+
+interface UseGroupLocks {
+  recoverable: BalanceLock[]
+  unRecoverable: BalanceLock[]
+}
+
+export const useGroupLocks = (address: string, locks: BalanceLock[]): UseGroupLocks => {
+  const { unstakingPeriodEnd, latestElection, voteElection, application, vote } =
```

Comment:
Now I get why the logic was split between a hook and helper functions in the first place.

This is better now, but this could be avoided by having this structure:
```ts
const hasWGLock = locks.some(({ type }) => type.endsWith('Worker'))
const hasCandidateLock = locks.some(({ type }) => type === 'Council Candidate')
const hasVoteLock = locks.some(({ type }) => type === 'Voting')

const isWGLockRecoverable = useIsWGLockRecoverable(hasWGLock, adress)
const isCandidateLockRecoverable = useIsCandidateLockRecoverable(hasCandidateLock, address)
const isVoteLockRecoverable = useIsVoteLockRecoverable(hasVoteLock, address)

return locks.reduce( ...
```

This way the 1st parameter of `useIsXXXLockRecoverable` is a `!skip` which prevent the queries to be send for every accounts. And also all the logic for a given lock is in its specific hook.